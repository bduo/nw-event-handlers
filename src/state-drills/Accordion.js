import React, { Component } from 'react';
import './Accordion.css';

class Accordion extends Component {
    static defaultProps = {
        sections: [],
    };
    
    state = {
        currentSectionIndex: null
    }

    handleButtonClick(index) {
        console.log('button clicked!')
        this.setState({
            currentSectionIndex: index
        })
    }

    renderButtons = () => {
        return this.props.sections.map((sections, index) => (
          <li key={index} >
            <button onClick={() => this.handleButtonClick(index)}>
                {sections.title}
            </button>
            {(index === this.state.currentSectionIndex) && 
            <p>{sections.content}</p>}
          </li>
        ))
    }

    render() {
        return (
            <div>
                <ul>
                   {this.renderButtons()}
                </ul>
            </div>
    )}
}

export default Accordion