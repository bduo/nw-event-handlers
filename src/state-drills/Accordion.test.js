import React, {Component} from 'react';
import Accordion from './Accordion.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe(`Accordion Component`, () => {
    const sections = [
        {
          title: 'Sloth',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Lemur',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Scorpion',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]

    it('component renders empty li when sections prop is not given or supplied', () => {
        const wrapper = shallow(<Accordion />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('component renders no sections as active by default', () => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('component opens a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        wrapper.find('button').at(0).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('The component only opens the last section when multiple sections have been clicked.', () => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})