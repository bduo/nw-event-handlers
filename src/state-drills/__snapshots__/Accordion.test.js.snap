// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component component opens a clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Sloth
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Lemur
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Scorpion
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component component renders empty li when sections prop is not given or supplied 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordion Component component renders no sections as active by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Sloth
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Lemur
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Scorpion
      </button>
    </li>
  </ul>
</div>
`;
